### PR TITLE
Add Odo Project command.

### PR DESCRIPTION
**Pull request by dev-gaur**
fixes https://github.com/fabric8io/fabric8-maven-plugin/issues/1508

### CODE REVIEW

File: core/src/main/java/com/lordofthejars/odo/core/commands/ProjectCommand.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        this.arguments.add(COMMAND_NAME);
+    }
+
+    @Override
+    public List<String> getCliCommand() {
+        if (this.arguments.size() >= 3) {
+            return this.arguments;
+        }
+
+        final List<String> arguments = new ArrayList<>();
```
Comment (lordofthejars) [inline core/src/main/java/com/lordofthejars/odo/core/commands/ProjectCommand.java]: Here you are overriding anything inside `arguments`

File: core/src/main/java/com/lordofthejars/odo/core/commands/ProjectCreateCommand.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package com.lordofthejars.odo.core.commands;
+
+import com.lordofthejars.odo.api.Command;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProjectCreateCommand implements Command {
+    private static final String COMMAND_PREFIX = "project";
```
Comment (lordofthejars) [inline core/src/main/java/com/lordofthejars/odo/core/commands/ProjectCreateCommand.java]: Prefix should not be set here, it should be added in the parent command. See https://github.com/lordofthejars/odo-java/blob/master/core/src/main/java/com/lordofthejars/odo/core/commands/AppCommand.java#L29

File: core/src/main/java/com/lordofthejars/odo/core/commands/ProjectDeleteCommand.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package com.lordofthejars.odo.core.commands;
+
+import com.lordofthejars.odo.api.Command;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProjectDeleteCommand implements Command {
+    private static final String COMMAND_PREFIX = "project";
```
Comment (lordofthejars) [inline core/src/main/java/com/lordofthejars/odo/core/commands/ProjectDeleteCommand.java]: Same as before

File: core/src/main/java/com/lordofthejars/odo/core/commands/ProjectCommand.java (hunk 1 above)
Comment (dev-gaur) [reply core/src/main/java/com/lordofthejars/odo/core/commands/ProjectCommand.java]: ```this.arguments``` is returned even before ```arguments``` variable is created.

still, I'll change it for clarity's sake.

File: core/src/main/java/com/lordofthejars/odo/core/commands/ProjectCreateCommand.java (hunk 2 above)
Comment (dev-gaur) [reply core/src/main/java/com/lordofthejars/odo/core/commands/ProjectCreateCommand.java]: oh. got it. check now.

**Pull request merged**